packages/react/src/hooks/useFocusTrap.ts

```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import {focusTrap} from '@primer/behaviors'
+import {useProvidedRefOrCreate} from './useProvidedRefOrCreate'
+
+export interface FocusTrapHookSettings {
+  /**
+   * Ref that will be used for the trapping container. If not provided, one will
+   * be created by this hook and returned.
+   */
+  containerRef?: React.RefObject<HTMLElement>
+
+  /**
+   * Ref for the element that should receive focus when the focus trap is first enabled. If
+   * not provided, one will be created by this hook and returned. Its use is optional.
+   */
+  initialFocusRef?: React.RefObject<HTMLElement>
+
+  /**
+   * Set to true to disable the focus trap and clean up listeners. Can be re-enabled at
+   * any time.
+   */
+  disabled?: boolean
+
+  /**
+   * If true, when this focus trap is cleaned up, restore focus to the element that had
+   * focus immediately before the focus trap was enabled. (Default: false)
+   */
+  restoreFocusOnCleanUp?: boolean
+}
+
+/**
+ * Hook used to trap focus inside a container. Returns a ref that can be added to the container
+ * that should trap focus.
+ * @param settings {FocusTrapHookSettings}
+ */
+export function useFocusTrap(
```

Comment:
Are there any tests for this hook that we could also copy over here?

Reply:
Doesn't seem like there is. Only a story which is dependent on some PRC components. I can definitely follow up with a new test for this hook after this PR if that's okay?

Reply:
Yeah that'd be great, thanks. No rush.